### PR TITLE
Fix inconsistent discovery

### PR DESCRIPTION
**Pull request by danielrbradley**
- Iterate versions and spec paths in the same order every time.
- Leaving in the overwriting behaviour for now as exiting on a duplicate causes breaking changes in the schema.

### CODE REVIEW

File: provider/pkg/openapi/discover.go [hunk 1]
```diff
@@ -104,10 +106,18 @@ func ReadVersions(namespace string) AzureProviders {
 
 		// Set compatible versions to all other versions of the resource with the same normalized API path.
 		pathVersions := calculatePathVersions(versionMap)
-		for version, items := range versionMap {
+		versions := []string{}
+		for version := range versionMap {
+			versions = append(versions, version)
+		}
+		sort.Strings(versions)
+		for version := range versionMap {
```
Comment (mikhailshilkov) [inline provider/pkg/openapi/discover.go:114]: Shouldn't this iterate on `versions`?

**Review by mikhailshilkov — APPROVED** [review]
Makes sense to me overall - I think you missed one sorting. Then we can 🚢

File: provider/pkg/openapi/discover.go (hunk 1 above)
Comment (danielrbradley) [reply provider/pkg/openapi/discover.go:114]: Oops - yes - must have messed it up when testing turning these on and off 🤦

File: provider/pkg/openapi/discover.go [hunk 2]
```diff
@@ -305,6 +323,9 @@ func addAPIPath(providers AzureProviders, fileLocation, path string, swagger *Sp
 		case pathItem.Head != nil && !pathItem.Head.Deprecated:
 			typeName := resources.ResourceName(pathItem.Head.ID)
 			if typeName != "" && hasDelete {
+				if _, ok := version.Resources[typeName]; ok && version.Resources[typeName].Path != path {
+					fmt.Printf("warning: duplicate resource with type name %s at paths:\n  - %s\n  - %s\n", typeName, path, version.Resources[typeName].Path)
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/openapi/discover.go:327]: Yes saw this one too - it's probably not a big deal being just a case change. The full set of overwrites are also tracked in this issue: https://github.com/pulumi/pulumi-azure-native/issues/2149

File: provider/pkg/openapi/discover.go [hunk 3]
```diff
@@ -346,6 +370,9 @@ func addAPIPath(providers AzureProviders, fileLocation, path string, swagger *Sp
 		defaultBody, hasDefault := defaultResourcesStateNormalized[normalizePath(path)]
 		typeName := resources.ResourceName(pathItem.Get.ID)
 		if typeName != "" && hasDefault {
+			if _, ok := version.Resources[typeName]; ok && version.Resources[typeName].Path != path {
+				fmt.Printf("warning: duplicate resource with type name %s at paths:\n  - %s\n  - %s\n", typeName, path, version.Resources[typeName].Path)
```
[the remark this answers is not in the document]
Comment (danielrbradley) [reply provider/pkg/openapi/discover.go]: Unfortunately that will only tell us one half of the method pair being used. Have updated to include API version to at least let us find the exact spec with the issue.

**Pull request merged**